WeatherData.py

```diff
@@ -0,0 +1,47 @@
+# import re
+class WeatherData:
+    """This Data class is used to load the data from files and populate them in lists for processing"""
+    def __init__(self):
+        """Constructor id used for initializing two lists"""
+        self.names = []
+        self.values = []
+
+    def type_caster(self, value):
+        """type_caster is used for type cast any variable to float if possible
+
+        Args:
+            value ([any]): [variable send for typecasting]
+
+        Returns:
+            [float | None]: [ as per conditions ]
+        """    
+        try:
+            return float(value)
+        except:
+            return None
+    
+    def populate_values(self, search_able_files, folder_dir):
+        """populate_values is used for read the required files and populate them in lists
+        Args:
+            search_able_files ([str]): [all file names that is going to be searched]
```

Comment:
this method is accepting two arguments but in doc string you just mentioned single

also searchable is a single word. no need to add underscore between it